components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -274,24 +275,25 @@ private String processLogoutRequest(HttpServletRequest request, HttpServletRespo
     /**
      * Validate Id token signature
      * @param idToken Id token
+     * @param tenantDomain tenant Domain
      * @return validation state
      */
-    private boolean validateIdToken(String idToken) {
+    private boolean validateIdToken(String idToken, String tenantDomain) {
 
-        String tenantDomain = getTenantDomainForSignatureValidation(idToken);
-        if (StringUtils.isEmpty(tenantDomain)) {
+        String tenantDomainForSignatureValidation = getTenantDomainForSignatureValidation(idToken, tenantDomain);
+        if (StringUtils.isEmpty(tenantDomainForSignatureValidation)) {
             return false;
         }
-        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomainForSignatureValidation);
         RSAPublicKey publicKey;
 
         try {
             KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(tenantId);
 
-            if (!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
-                String ksName = tenantDomain.trim().replace(".", "-");
+            if (!tenantDomainForSignatureValidation.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
```

Comment:
can tenantDomainForSignatureValidation be null. Shall we swap the constant with variable?

Reply:
fixed